3737/1/parser.rb

```diff
@@ -0,0 +1,98 @@
+require 'nokogiri'
+require 'open-uri'
+require 'xpath'
+
+# class for parsing info about gem from github page
+class Parser
+  def initialize
+    @arr = {}
+  end
+
+  def repository(gname)
+    link = search_repo(gname)
+    return unless link
+    gethash(link)
+  end
+
+  private
+
+  def search_repo(gname)
+    url = "http://rubygems.org/gems/#{gname}"
+    check_page('code', url) || check_page('home', url)
+  end
+
+  def check_page(linkid, url)
+    link = Nokogiri::HTML(URI.open(url)).css("//a[@id=#{linkid}]").attr('href')
+    return unless link
+    link.text if link.text.include?('github')
+  end
+
+  def git_repo(link)
+    github_page = Nokogiri::HTML(URI.open(link))
```

Comment:
no need in this variable and next line. ruby method returns the value of last calculated expression